pkgs/native_assets_cli/lib/src/api/build_config.dart

```diff
@@ -138,29 +137,29 @@ abstract final class BuildConfig {
   /// If not provided, [workingDirectory] defaults to [Directory.current].
   ///
   /// This async constructor is intended to be used directly in CLI files.
-  static Future<BuildConfig> fromArguments(
-    List<String> args, {
+  factory BuildConfig(
+    List<String> arguments, {
     Map<String, String>? environment,
     Uri? workingDirectory,
   }) =>
-      BuildConfigImpl.fromArgs(
-        args,
+      BuildConfigImpl.fromArguments(
+        arguments,
         environment: environment,
         workingDirectory: workingDirectory,
       );
 
   /// Constructs a config for a non-dry run by providing values for each field.
   ///
-  /// `build.dart` hooks will most likely use [BuildConfig.fromArguments].
-  /// However, for unit testing code which consumes a [BuildConfig], this
-  /// constructor facilitates easy construction.
+  /// `build.dart` hooks will most likely use the unnamed constructor. However,
```

Comment:
I thought `build.dart` will not use a constructor at all, but the `build` method taking a function as input?